Makefile

```
@@ -0,0 +1,76 @@
+# variables to store the names used
+NAME_MY_PROJECT = minishell
+NAME_LIBFT = libft.a
+GREEN = \033[32m
+RESET = \033[0m
+
+# directories
+MY_LIBFT_DIR = libft
+MY_SRCS_DIR = srcs
+CUR_DIR = $(shell pwd)
+
+# files used
+MY_LIBFT_SRCS = $(addprefix $(MY_LIBFT_DIR)/, \
+				ft_split.c \
+				ft_strdup.c \
+				ft_strlen.c \
+				ft_strlcpy.c \
+				ft_strchr.c \
+				ft_substr.c)
+
+MY_PROJECT_SRCS = $(addprefix $(MY_SRCS_DIR)/, \
+				minishell.c \
+				get_token.c \
+				token_functions.c)
+
+# o files used
+MY_LIBFT_OBJS = $(MY_LIBFT_SRCS:.c=.o)
+MY_PROJECT_OBJS = $(MY_PROJECT_SRCS:.c=.o)
+
+# c flags used during compilation
+CFLAGS = -Wall -Wextra -Werror -g
```

```suggestion
CFLAGS = -Wall -Wextra -g3 -fsanitize=address,undefined
```

Je recommande ces CFLAGS pour la phase de développement.
* -g3 c'est pareil que -g mais avec + d'infos
* -fsanitize=address ca permet de détecter les "memory leaks" sans meme d'utiliser valgrind
* -fsanitize=undefined ca permet de détecter les "undefined behaviors"
* -Werror c'est chiant parce que ca empêche de compiler si y'a une petite erreur